### PR TITLE
*: replace part of path with filepath

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #8475

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/grpcutil/grpcutil_test.go [hunk 1]
```diff
@@ -41,9 +41,9 @@ func TestToTLSConfig(t *testing.T) {
 
 	re := require.New(t)
 	tlsConfig := TLSConfig{
-		KeyPath:  path.Join(certPath, "pd-server-key.pem"),
-		CertPath: path.Join(certPath, "pd-server.pem"),
-		CAPath:   path.Join(certPath, "ca.pem"),
+		KeyPath:  filepath.Join(certPath, "pd-server-key.pem"),
```
Comment (HuSharp) [inline pkg/utils/grpcutil/grpcutil_test.go:44]: Why change only `a part of` it? It looks like there's still quite a bit of `path` used

Comment (rleungx) [reply pkg/utils/grpcutil/grpcutil_test.go:44]: some of them are etcd keys which doesn't rely on os

Comment (HuSharp) [reply pkg/utils/grpcutil/grpcutil_test.go:44]: got it.

**Review by HuSharp — APPROVED** [review]

File: tools/pd-ut/ut.go [hunk 2]
```diff
@@ -746,13 +746,13 @@ func buildTestBinaryMulti(pkgs []string) ([]byte, error) {
 
 	// go test --exec=xprog --tags=tso_function_test,deadlock -vet=off --count=0 $(pkgs)
 	// workPath just like `/pd/tests/integrations`
-	xprogPath := path.Join(workDir, "bin/xprog")
+	xprogPath := filepath.Join(workDir, "bin/xprog")
```
Comment (HuSharp) [inline tools/pd-ut/ut.go:749]: If we consider different operating systems, maybe`bin/xprog` also need to be changed?

**Review by nolouch — APPROVED** [review]

**Pull request merged**